### PR TITLE
python3Packages.pystac-client: init at 0.8.3

### PR DESCRIPTION
**Pull request by imincik**
## Description of changes

Add pystac-client package at version 0.8.3.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pystac-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    pytest-benchmark
+    pytestCheckHook
+    pytest-console-scripts
+    pytest-mock
+    pytest-recording
+    requests-mock
+  ];
+
+  disabledTests = [
+    # tests are accessing network
```
Comment (l0b0) [inline pkgs/development/python-modules/pystac-client/default.nix:50]: Is it possible to filter for tests marked `vcr`, that is, decorated with `@pytest.mark.vcr`? That should simplify the list, and clarify whether any tests are missing a marker to indicate that they need Internet access to work.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -10501,6 +10501,8 @@ self: super: with self; {
 
   pystac = callPackage ../development/python-modules/pystac { };
 
+  pystac-client = callPackage ../development/python-modules/pystac-client { };
```
Comment (l0b0) [inline pkgs/top-level/python-packages.nix:10504]: Should this package be in `pkgs/by-name`? I'm not sure what the guideline is for sub-packages, but this file calls at least one by-name package.

**Review by l0b0 — APPROVED** [review]
Minor comments only. Untested.

Thanks!

File: pkgs/development/python-modules/pystac-client/default.nix (hunk 1 above)
Comment (imincik) [reply pkgs/development/python-modules/pystac-client/default.nix:50]: Good idea. Done now.

**Comment by natsukium** [comment]
result of `nixpkgs-lint --include-unfinished-lints pkgs/development/python-modules/pystac-client` [1](https://github.com/natsukium/nixpkgs-lint/tree/620731fc7ec66b0c67e6b8ea039545f71ea78409)
```
Advice: starts with definite or indefinite article in description
    ╭─[pkgs/development/python-modules/pystac-client/default.nix:79:19]
    │
 79 │     description = "A Python client for working with STAC Catalogs and APIs";
    │                   ────────────────────────────┬────────────────────────────  
    │                                               ╰────────────────────────────── remove a definite/indefinite article from meta.description
────╯
Advice: pytest-benchmark in nativeCheckInputs
    ╭─[pkgs/development/python-modules/pystac-client/default.nix:41:5]
    │
 40 │ ╭─▶   nativeCheckInputs = [
 41 │ │       pytest-benchmark
    │ │       ────────┬───────  
    │ │               ╰───────── remove this from nativeCheckInputs or pass `--benchmark-disable` to pytestFlagsArray
    ┆ ┆   
 47 │ ├─▶   ];
    │ │          
    │ ╰────────── part of this list
────╯
```

File: pkgs/development/python-modules/pystac-client/default.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  src = fetchFromGitHub {
+    owner = "stac-utils";
+    repo = "pystac-client";
+    rev = "v${version}";
+    hash = "sha256-tzfpvNtj+KkKjA75En+OwxYQWGzxHLACLkzWT2j/ThU=";
+  };
+
+  build-system = [ setuptools ];
+
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/pystac-client/default.nix:34]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/pystac-client/default.nix [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    pystac
+    python-dateutil
+    requests
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pystac-client/default.nix:38]:
```suggestion
  ] ++ pystac.optional-dependencies.validation;
```
https://github.com/stac-utils/pystac-client/blob/b65b325d31019025faa17b32584c81967cbcc9e2/pyproject.toml#L29-L33

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/top-level/python-packages.nix:10504]: The python package must be added to python-packages.nix to be called correctly.

File: pkgs/development/python-modules/pystac-client/default.nix (hunk 4 above)
Comment (imincik) [reply pkgs/development/python-modules/pystac-client/default.nix]: This doesn't work. 

```
       error: attribute 'optional-dependencies' missing

       at /home/imincik/Projects/nix/nixpkgs/code/pkgs/development/python-modules/pystac-client/default.nix:38:8:

           37|     requests
           38|   ] ++ pystac.optional-dependencies.validation;
             |        ^
           39|
```

**Comment by imincik** [comment]
> Advice: pytest-benchmark in nativeCheckInputs
>     ╭─[pkgs/development/python-modules/pystac-client/default.nix:41:5]
>     │
>  40 │ ╭─▶   nativeCheckInputs = [
>  41 │ │       pytest-benchmark
>     │ │       ────────┬───────  
>     │ │               ╰───────── remove this from nativeCheckInputs or pass `--benchmark-disable` to pytestFlagsArray
>     ┆ ┆   
>  47 │ ├─▶   ];
>     │ │          
>     │ ╰────────── part of this list
> ────╯

Removal of `pytest-benchmark` is causing following error:

```
       > Executing pytestCheckPhase
       > ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
       > __main__.py: error: unrecognized arguments: --benchmark-skip
       >   inifile: /build/source/pyproject.toml
       >   rootdir: /build/source
```

`--benchmark-skip` is used in `pyproject.toml`.

**Review by sikmir — APPROVED** [review]

**Pull request merged**